### DISCUSSION TITLE
why m_nerf.training.optimize_{}s variables are all fasle

### DISCUSSION THREAD

**Discussion by neilgogogo** [NVlabs/instant-ngp#181] [Q&A] [answered]
I am sorry I am a little confused with optimize_{}s variables. Why the optimize_{}s are all false, and will the training result be better if there are all true.

https://github.com/NVlabs/instant-ngp/blob/3d15eb48a5e96633d40f1bf65c1c3024bc0a208c/include/neural-graphics-primitives/testbed.h#L489-L492

Kind regards

**Comment by Tom94** [comment, marked as answer]
Hi there,

these variables (in particular `optimize_distortion` and `optimize_extrinsics`) can help with training real-world footage that has suboptimally estimated camera parameters (such as from an online SLAM algorithm).

For all our paper results, we used either synthetic data (known camera parameters) or data with very good camera parameter fits (through offline COLMAP). Thus, fine-tuning of the camera parameters rarely gave a meaningful improvement and wasn't worth the extra compute.

That's why these are off by default -- but they can help a lot when your training data has slightly incorrect poses or lens distortion params.
